### PR TITLE
fix spectral extraction previews on unit change

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request updates the spectral extraction live-preview for any change that affects the y-units of the extracted spectrum or spectral viewer (and also ensures the internal units in the plugin are populated for the warning banner introduced in #3143 which was recently broken).

https://github.com/user-attachments/assets/c4ec1fd9-8cdf-449f-873e-d84b6a8a212a

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py [hunk 1]
```diff
@@ -109,8 +110,8 @@ class SpectralExtraction(PluginTemplateMixin, ApertureSubsetSelectMixin,
 
     results_units = Unicode().tag(sync=True)
     spectrum_y_units = Unicode().tag(sync=True)
-    flux_unit = Unicode().tag(sync=True)
-    sb_unit = Unicode().tag(sync=True)
+    flux_units = Unicode().tag(sync=True)
+    sb_units = Unicode().tag(sync=True)
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:114]: just to be consistent with the traitlets above, I was confusing myself when writing code later in the plugin and having to switch between singular and plural

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py [hunk 2]
```diff
@@ -542,7 +545,9 @@ def _preview_x_from_extracted(self, extracted):
         return extracted.spectral_axis.value
 
     def _preview_y_from_extracted(self, extracted):
-        return extracted.flux.value
+        # TODO: use extracted's PIXAR_SR instead (but for some reason isn't populated here...)
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:548]: does anyone know why this is (I'm guessing its just populated later when actually loading into the app)?  Ultimately, I think the extracted spectrum just gets a copy of PIXAR_SR from the input cube, so this shouldn't matter, it just would read more cleanly if it did.

**Comment by kecnry** [comment]
moving back to draft - the test failures are all related to non per-steradian SB units, which will be addressed by #3156.  Instead of writing a temporary workaround, feel free to test/review the diff and we can rebase this once #3156 is in.

**Comment by rosteen** [comment]
It seems like changing to flux makes the preview disappear (as expected, I think), but then changing back to surface brightness doesn't trigger the preview to reappear. It only comes back if you then change the flux unit while in surface brightness mode.

**Comment by kecnry** [comment]
is it possible that its timing out?  Do you get the warning that the previews have been disabled?

**Comment by rosteen** [comment]
The previews disabled warning does not appear, I do see the warning about the display units (surface brightness vs flux). And now that I think about it, I guess the preview should show up when flux is selected in unit conversion. It doesn't seem to be timing out.

I was able to get the preview to show this time. You can see that at first it flickered and then disappeared when I switched to flux, and when I changed units (the second time?) it stuck around:

https://github.com/user-attachments/assets/1ea09c66-ca8e-42cd-9a36-e235f5bf9b67

**Review by cshanahan1 — COMMENTED** [review]
If you load in a surface brightness cube, the automatic sum extraction is plotted in the spectral viewer and the spectral y axis correctly says flux, however in the unit conversion app the toggle is set to 'surface brightness', i don't see this on main

**Comment by kecnry** [comment]
@cshanahan1 - I think this is fixed now after a rebase, but please confirm if you have a second to test.

**Comment by kecnry** [comment]
(still working on some cases where the preview is not showing after the rebase)

**Review by cshanahan1 — APPROVED** [review]
seems to work as intended - I can't convert to erg/s/cm2/a but i think that will be fixed by mine and @pllim PRs

**Review by rosteen — APPROVED** [review]
Works for me now, thanks.

**Pull request merged**